apps/backend/src/location/location.service.ts

```diff
@@ -1,86 +1,107 @@
-        const key = itm.location_id;
+
+  public async seedMinistryLocations(
+    locations: MasterLocationEntity[],
+    program: Ministries
+  ) {
+    // master location data list contains multiple rows for each location_id
+    // split into location table (FK to transaction table), location_merchant table (FK to location_id && pos-deposit), and location_bank table (FK to location_id && cash_deposit)
+    const locationEntityList = locations.reduce(
+      (
+        acc: { [key: string]: Partial<LocationEntity> },
+        itm: MasterLocationEntity
+      ) => {
+        const key = `${itm.location_id}${itm.source_id}`;
         if (!acc[key]) {
           acc[key] = {
             source_id: itm.source_id,
             location_id: itm.location_id,
             program_code: itm.program_code,
+            program_desc: itm.program_desc,
             ministry_client: itm.ministry_client,
             resp_code: itm.resp_code,
             service_line_code: itm.service_line_code,
             stob_code: itm.stob_code,
             project_code: itm.project_code,
-            merchant_ids: [],
-            pt_location_id: 0,
+            banks: [],
+            merchants: [],
             description: '',
           };
         }
+
         itm.merchant_id !== BankMerchantId &&
-          acc[key].merchant_ids.push(itm.merchant_id);
-        if (itm.method === LocationMethod.Bank) {
-          acc[key].pt_location_id = itm.pt_location_id;
+          !acc[key].merchants?.find((merch) => merch.id === itm.merchant_id) &&
+          acc[key].merchants?.push(
+            new MerchantLocationEntity({ id: itm.merchant_id })
+          );
+
+        !acc[key].banks?.find((pt) => pt.id === itm.pt_location_id) &&
+          acc[key].banks?.push(
+            new BankLocationEntity({ id: itm.pt_location_id })
+          );
+        if (program === 'SBC') {
+          acc[key].description = locations.find(
+            (loc) =>
+              loc.location_id === itm.location_id && loc.method === 'Bank'
+          )!.description;
+        } else {
           acc[key].description = itm.description;
         }
+
         return acc;
       },
       {}
     );
-    return Object.values(normalizedLocationList);
+    await this.ministryLocationRepo.save(
+      this.ministryLocationRepo.create(
+        Object.values(locationEntityList).map((itm) => new LocationEntity(itm))
+      )
+    );
   }
 
-  public async getLocationsBySource(
-    source: Ministries
-  ): Promise<NormalizedLocation[]> {
-    const locations = await this.locationRepo.find({
-      where: {
-        source_id: source,
-      },
-      order: {
-        location_id: 'ASC',
-      },
-    });
-    return this.normalizeLocations(locations);
+  public async createLocations(
+    locationsData: MasterLocationEntity[]
+  ): Promise<void> {
+    await this.locationRepo.save(this.locationRepo.create(locationsData));
   }
 }
```

Comment:
I will ask you to do a readability pass on this file again with the thought of an unfamiliar dev looking at the functions. Function comments will help, but finding a place to document how a location splits into a location / bank / merchant and master location would be a good thing to do imo